### PR TITLE
Include missing properties on Exporter Settings

### PR DESCRIPTION
**Pull request by colin-higgins**
Introduce tests for `ImmutableExporterSettings` similar to `ImmutableTracerSettings`

@DataDog/apm-dotnet

### CODE REVIEW

File: tracer/src/Datadog.Trace/Configuration/ImmutableExporterSettings.cs [hunk 1]
```diff
@@ -29,11 +29,13 @@ public ImmutableExporterSettings(IConfigurationSource source)
         /// <param name="settings">The tracer settings to use to populate the immutable tracer settings</param>
         public ImmutableExporterSettings(ExporterSettings settings)
         {
-            TraceTransport = settings.TracesTransport;
             AgentUri = settings.AgentUri;
+            TraceTransport = settings.TracesTransport;
```
Comment (zacharycmontoya) [inline tracer/src/Datadog.Trace/Configuration/ImmutableExporterSettings.cs:33]: I think this should use the internal type and you can remove the public TraceTransport
```suggestion
            TracesTransport = settings.TracesTransport;
```

**Review by zacharycmontoya — APPROVED** [review]
LGTM

File: tracer/src/Datadog.Trace/Configuration/ImmutableExporterSettings.cs [hunk 2]
```diff
@@ -70,5 +72,33 @@ public ImmutableExporterSettings(ExporterSettings settings)
...
+
+        /// <summary>
+        /// Gets the minimum number of closed spans in a trace before it's partially flushed
+        /// </summary>
+        public int PartialFlushMinSpans { get; }
+
+        /// <summary>
+        /// Gets the transport used to send traces to the Agent.
+        /// </summary>
+        internal TracesTransportType TracesTransport { get; }
```
Comment (lucaspimentel) [inline tracer/src/Datadog.Trace/Configuration/ImmutableExporterSettings.cs:96]: `TracesTransport` (with plural `Traces`) sounds weird to me. Shouldn't that be `TraceTransportType` and `TraceTransport`?

**Review by lucaspimentel — APPROVED** [review]

File: tracer/src/Datadog.Trace/Configuration/ImmutableExporterSettings.cs (hunk 2 above)
Comment (colin-higgins) [reply tracer/src/Datadog.Trace/Configuration/ImmutableExporterSettings.cs:96]: I'm just trying to be consistent here, everything else is plural. 😆

**Comment by andrewlock** [comment]
## Code Coverage Report :bar_chart:

:heavy_check_mark: Merging #2179 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/d939f81bdfbf85272b349ca1899590d2b76b2707) will **not change** line coverage
:heavy_check_mark: Merging #2179 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/d939f81bdfbf85272b349ca1899590d2b76b2707) will **not change** branch coverage
:heavy_check_mark: Merging #2179 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/d939f81bdfbf85272b349ca1899590d2b76b2707) will will **decrease** complexity by `1`


|           | [master](https://github.com/DataDog/dd-trace-dotnet/tree/d939f81bdfbf85272b349ca1899590d2b76b2707) | #2179       | Change   | 
|:----------|:-----------:|:-----------:|:--------:|
| Lines     | `9229` / `12688` | `9218` / `12690` |          |
| Lines %   | `73%`      | `73%`      |  `0%` :heavy_check_mark:  |
| Branches  | `4207` / `6192` | `4198` / `6192` |          |
| Branches %| `68%`      | `68%`      |  `0%` :heavy_check_mark:  |
| Complexity|   `7370`      | `7369`        |  `-1`  :heavy_check_mark:    |

View the full report for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=84081&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/84081/artifacts?artifactName=Code%20Coverage%20Report_84081&api-version=6.0&%24format=zip)  
* [HTML report for this PR #2179](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=84111&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/84111/artifacts?artifactName=Code%20Coverage%20Report_84111&api-version=6.0&%24format=zip)

### Datadog.Trace Breakdown :heavy_check_mark:

|        | [master](https://github.com/DataDog/dd-trace-dotnet/tree/d939f81bdfbf85272b349ca1899590d2b76b2707) | #2179       | Change   | 
|:-------|:-----------:|:-----------:|:--------:|
| Lines %| `73%`      | `73%`       |  `0%` :heavy_check_mark:  |
| Branches %| `68%`      | `68%`       |  `0%` :heavy_check_mark:  |
| Complexity| `7370`      | `7369`       |  `-1` :heavy_check_mark:  |

The following classes have significant coverage changes.

| File    | Line coverage change | Branch coverage change | Complexity change |
|:--------|:--------------------:|:----------:|:--------:|
| [Datadog.Trace.ClrProfiler.AutoInstrumentation.Couchbase.IIOServiceExecuteIntegrationBis](https://github.com/DataDog/dd-trace-dotnet/tree/1f668358617274086017547ace35a37d8da44295/tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Couchbase/IIOServiceExecuteIntegrationBis.cs) | `-100%` :no_entry: | `0%` :heavy_check_mark:   | `0` :heavy_check_mark: |
| [Datadog.Trace.ClrProfiler.AutoInstrumentation.Testing.MsTestV2.TestMethodRunnerExecuteIntegration](https://github.com/DataDog/dd-trace-dotnet/tree/1f668358617274086017547ace35a37d8da44295/tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Testing/MsTestV2/TestMethodRunnerExecuteIntegration.cs) | `-33%` :no_entry: | `-19%` :no_entry:   | `0` :heavy_check_mark: |
| [Datadog.Trace.ClrProfiler.AutoInstrumentation.Testing.NUnit.NUnitCompositeWorkItemSkipChildrenIntegration](https://github.com/DataDog/dd-trace-dotnet/tree/1f668358617274086017547ace35a37d8da44295/tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Testing/NUnit/NUnitCompositeWorkItemSkipChildrenIntegration.cs) | `-8%` :no_entry: | `-8%` :no_entry:   | `0` :heavy_check_mark: |
| [Datadog.Trace.Configuration.ImmutableExporterSettings](https://github.com/DataDog/dd-trace-dotnet/tree/1f668358617274086017547ace35a37d8da44295/tracer/src/Datadog.Trace/Configuration/ImmutableExporterSettings.cs) | `8%` :heavy_check_mark: | `0%` :heavy_check_mark:   | `0` :heavy_check_mark: |


View the full reports for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=84081&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/84081/artifacts?artifactName=Code%20Coverage%20Report_84081&api-version=6.0&%24format=zip)  
* [HTML report for this PR #2179](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=84111&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/84111/artifacts?artifactName=Code%20Coverage%20Report_84111&api-version=6.0&%24format=zip)

**Pull request merged**